src/software/world/refbox_constants.cpp

```diff
@@ -1,51 +1,11 @@
 #include "software/world/refbox_constants.h"
 
 #include <map>
-
-static const std::map<RefboxGameState, std::string> refbox_game_state_names = {
-    {RefboxGameState::HALT, "HALT"},
-    {RefboxGameState::STOP, "STOP"},
-    {RefboxGameState::NORMAL_START, "NORMAL_START"},
-    {RefboxGameState::FORCE_START, "FORCE_START"},
-    {RefboxGameState::PREPARE_KICKOFF_US, "PREPARE_KICKOFF_US"},
-    {RefboxGameState::PREPARE_KICKOFF_THEM, "PREPARE_KICKOFF_THEM"},
-    {RefboxGameState::PREPARE_PENALTY_US, "PREPARE_PENALTY_US"},
-    {RefboxGameState::PREPARE_PENALTY_THEM, "PREPARE_PENALTY_THEM"},
-    {RefboxGameState::DIRECT_FREE_US, "DIRECT_FREE_US"},
-    {RefboxGameState::DIRECT_FREE_THEM, "DIRECT_FREE_THEM"},
-    {RefboxGameState::INDIRECT_FREE_US, "INDIRECT_FREE_US"},
-    {RefboxGameState::INDIRECT_FREE_THEM, "INDIRECT_FREE_THEM"},
-    {RefboxGameState::TIMEOUT_US, "TIMEOUT_US"},
-    {RefboxGameState::TIMEOUT_THEM, "TIMEOUT_THEM"},
-    {RefboxGameState::GOAL_US, "GOAL_US"},
-    {RefboxGameState::GOAL_THEM, "GOAL_THEM"},
-    {RefboxGameState::BALL_PLACEMENT_US, "BALL_PLACEMENT_US"},
-    {RefboxGameState::BALL_PLACEMENT_THEM, "BALL_PLACEMENT_THEM"},
-    {RefboxGameState::REFBOX_GAME_STATE_COUNT, "REFBOX_GAME_STATE_COUNT"}};
+#include <magic_enum/magic_enum.hpp>
 
 std::string name(const RefboxGameState& state)
 {
-    auto find_name_result = refbox_game_state_names.find(state);
-    if (find_name_result != refbox_game_state_names.end())
-    {
-        return find_name_result->second;
-    }
-    else
-    {
-        throw std::invalid_argument(
-            "Error: Tried to get the name of an invalid refbox gamestate");
-    }
-}
+    std::string find_name_result = static_cast<std::string>(magic_enum::enum_name(state));
 
-std::ostream& operator<<(std::ostream& os, const RefboxGameState& state)
-{
```

Comment:
Please keep this operator and have it call name()

Reply:
Added back. Also implemented for motion_constraints.cpp, where it is passing in the motion_constraints_manager_test.cpp.